File: basics/roscpp/simple_pub/src/simple_pub_iterative.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

int
main(int argc, char **argv)
{
	ros::init(argc, argv, "simple_pub_iterative_node");

	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::Int32>("/talk", 1);

	//2Hz rate

	ros::Rate loop_rate(2);

	std_msgs::Int32 count;
	count.data = 0;

	while(ros::ok()){
		pub.publish(count);
		ros::spinOnce();
		count.data++;
		loop_rate.sleep();
	}

	exit(EXIT_SUCCESS);
}

File: basics/rospy/simple_pub_py/src/simple_pub_iterative.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import Int32

def main():
    rospy.init_node("simple_pub_iterative_node")

    pub = rospy.Publisher("/talk", Int32, queue_size=1)

    msg = Int32()
    msg.data = 0

    # 2Hz Rate:

    rate = rospy.Rate(2)
    while(not rospy.is_shutdown()):
        pub.publish(msg)
        msg.data += 1
        rate.sleep()

if __name__ == "__main__":

    try:
        main()
    except rospy.ROSInterruptException:
        pass
